alf/algorithms/off_policy_algorithm.py

```diff
@@ -105,18 +101,41 @@ def _train_iter_off_policy(self):
         """User may override this for their own training procedure."""
         config: TrainerConfig = self._config
 
+        first_iter = (alf.summary.get_global_counter() == 0)
+
         if not config.update_counter_every_mini_batch:
             alf.summary.get_global_counter().add_(1)
 
-        with torch.no_grad():
```

Comment:
As a note, it will cost more memory usage without "with no_grad()".

Reply:
Yeah I know that. Potentially we could optimize this part by setting a flag indicating whether we should unroll with grads or not depending on the user's need for training sub-algorithms. But if `unroll_length` is small then it's probably not a big issue.